src/workers/python/PythonWorker.worker.ts

```diff
@@ -74,6 +74,22 @@ class PythonWorker extends Backend {
         }
     }
 
+    protected debugCodeInternal(code: string, breakpoints: Set<number>): Promise<any> {
+        if (breakpoints.size === 0) {
+            return this.runCodeInternal(code);
+        }
+        const pdbCode = code.split("\n").reduce(
+            (acc: Array<string>, current: string, currentIndex: number) => {
+                if (breakpoints.has(currentIndex)) {
+                    // extra lines for PDB to know when to halt
+                    acc.push("breakpoint()");
```

Comment:
This has many problems. For example it won't work if you set a breakpoint inside an indented block or most multiline expressions.

The base of `pdb` is `bdb` (https://docs.python.org/3/library/bdb.html) and it's an extensible, configurable framework, with support for breakpoints.

```python
import pdb
debugger = pdb.Pdb()
debugger.set_break(__file__, 8)
debugger.set_trace()
# press 'c' to continue...
def foo():
    x = 1
    print(x)  # should stop here (line 8)
    x += 2
    return x

foo()
```

Reply:
Indeed, I did not consider indentation just yet. Thanks for pointing this out, I'll use the described methods to implement it correctly.